src/TezosProvider.ts

```diff
@@ -0,0 +1,509 @@
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  // Method to send an undelegation
+  public tezosSendUndelegation = async (): Promise<TezosSendResponse> => {
+    const op: PartialTezosDelegationOperation = { kind: TezosOperationType.DELEGATION };
+    return await this.tezosSend(op);
+  };
+
+  // Method to originate a contract
+  public tezosSendOrigination = async (
+    op: PartialTezosOriginationOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  // Method to call a smart contract: destination is the contract address, entrypoint as defined in the contract
+  public tezosSendContractCall = async (
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendStake = async (
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return await this.tezosSend({
+      ...op,
+      destination: this.address,
+      parameters: { entrypoint: "stake", value: { prim: "Unit" } },
+    });
+  };
+
+  public tezosSendUnstake = async (
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return await this.tezosSend({
+      ...op,
+      destination: this.address,
+      parameters: { entrypoint: "unstake", value: { prim: "Unit" } },
+    });
+  };
+
+  public tezosSendFinalizeUnstake = async (
+    op: PartialTezosTransactionOperation,
+  ): Promise<TezosSendResponse> => {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return await this.tezosSend({
+      ...op,
+      destination: this.address,
+      parameters: { entrypoint: "finalize_unstake", value: { prim: "Unit" } },
+    });
+  };
+
+  public tezosSendActivateAccount = async (
+    op: TezosActivateAccountOperation,
+  ): Promise<TezosSendResponse> => {
+    if (!this.address) {
+      throw new TezosConnectionError();
+    }
+    return await this.tezosSend({ ...op, pkh: this.address });
+  };
+
+  public tezosSendBallot = async (op: TezosBallotOperation): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendDalPublishCommitment = async (
+    op: PartialTezosDalPublishCommitmentOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendFailingNoop = async (
+    op: TezosFailingNoopOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendIncreasePaidStorage = async (
+    op: PartialTezosIncreasePaidStorageOperation,
+  ): Promise<TezosSendResponse> => {
+    return await this.tezosSend(op);
+  };
+
+  public tezosSendProposal = async (op: TezosProposalOperation): Promise<TezosSendResponse> => {
```

Comment:
not sure if all those methods need the tezos prefix. it's obvious from the class name that it's tezos related

Reply:
The operations contain the names of WalletConnect requests: `tezos_send`, `tezos_sign`, `tezos_getaccounts`.
I try to find a way to distinguish requests to peer (on top of these 3) and auxilary requests, like `getBalance()`, `connect()`, `getContractAddress()`.
If you think, that it's not needed, I'll rename:
```
tezosSend -> send
tezosSign -> sign
tezosGetAccounts -> getAccounts
```

Reply:
yeah, I don't think it adds much value